ibllib/io/video.py

```diff
@@ -30,16 +30,19 @@ def get_video_frame(video_path, frame_number):
     return frame_image
 
 
-def get_video_frames_preload(video_path, frame_numbers, mask=Ellipsis, as_list=False):
+def get_video_frames_preload(video_path, frame_numbers=None, mask=Ellipsis, as_list=False,
+                             func=lambda x, **kwargs: x, **kwargs):
```

Comment:
Why are there two sets of kwargs here?  Because your passing in a function handle any constant arguments can be wrapped in a partial function instead.  For example `func=partial(cv2.cvtColor, code=cv2.COLOR_BGR2GRAY)` or just `func=lambda x: cv2.cvtColor(x, cv2.COLOR_BGR2GRAY)`

Reply:
Cool, didn't know about the partial function. Do you agree with having lambda x: x as a default for func? It gets around two if / else statements when applying func inside get_video_frames_preload